meilidb-http/src/data.rs

```diff
@@ -115,12 +91,10 @@ impl DataInner {
             .map(|(a, c)| (schema.attribute_name(a).to_owned(), c))
             .collect();
 
-        let key = format!("fields-frequency-{}", index_name);
-        self.db
-            .common_store()
-            .put::<Str, SerdeFreqsMap>(writer, &key, &frequency)?;
-
-        Ok(())
+        index
+            .main
+            .put_fields_frequency(&mut writer, &frequency)
```

Comment:
You do not need to make `writer` mutable and you should not give this function a `ref mut` of a `ref mut`.
```suggestion
            .put_fields_frequency(writer, &frequency)
```